src/journal-remote/journal-remote-main.c

```diff
@@ -98,10 +98,11 @@ static MHDDaemonWrapper* MHDDaemonWrapper_free(MHDDaemonWrapper *d) {
         if (!d)
                 return NULL;
 
+        d->io_event = sd_event_source_unref(d->io_event);
+        d->timer_event = sd_event_source_unref(d->timer_event);
```

Comment:
We don't clear individual fields if the struct is heap allocated and will be free'd anyways

Reply:
that's true. but MHD_stop_daemon() calls back into our callbacks, hence we need to make sure that we properly invalidated the two event sources before we call that, so that we know not to deref it anymore.